iothub/service/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -41,6 +28,15 @@ internal IotHubConnectionStringBuilder()
         {
         }
 
+        internal IotHubConnectionStringBuilder(string hostName, string sharedAccessKeyName, string sharedAccessKey, string sharedAccessSignature, string gatewayHostName)
```

Comment:
Is the SDK building a connection string for some scenario? If not, then we should require only a parser and not a builder. Generally, we'd expect the users to provide us with a formatted connection string that they would have simply copied off either portal or CLI.

Reply:
The SDK was not building connection string so removed the class.
The class was however being used in a couple e2e tests to retrieve host name value from the builder.
I updated the e2e tests to parse the conn string on the fly and fetch the host name value.